File: ThirdYear/prabhat-04_PrabhatKeshari_2125it1086_3/Week_2/RotateList.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(head==NULL)
            return head;
        ListNode *tail=head,*ptr=head;
        int n=0;
        while(tail->next!=NULL){
            n++;
            tail=tail->next;
        }
        n++;
        k=k%n;
        // cout<<k<<endl;
        if(k==0)
            return head;
        int cnt=1;
        while(cnt<n-k){
            cnt++;
            ptr=ptr->next;
        }
        ListNode*ans=ptr->next;
        // cout<<ans->val<<endl;
        ptr->next=NULL;
        tail->next=head;
        return ans;
    }
};

File: ThirdYear/harshi706_HarshiAgrawal_2125it1200_3/Week_2/rotateList.java
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || k == 0 || head.next == null)
            return head;
        ListNode previous = null;
        ListNode current = head;
        k = k % size(head);
        while (k > 0) {
            while(current.next != null) {
                previous = current;
                current = current.next;
            }
            previous.next = null;
            current.next = head;
            head = current;
            k--;
        }
        return head;
    }
    public int size(ListNode head) {
        int size = 0;
        while (head != null) {
            head = head.next;
            size++;
        }
        return size;
    }
}